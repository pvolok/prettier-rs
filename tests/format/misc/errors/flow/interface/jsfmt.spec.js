run_spec(
  {
    importMeta: import.meta,
    snippets: [
      // Not allowed
      "declare export interface A {}",

      // Can't have mixins nor implements
      "interface A mixins B {}",
      "interface A implements B {}",
      "declare interface A mixins B {}",
      "declare interface A implements B {}",
    ],
  },
  ["flow", "babel-flow"],
);
