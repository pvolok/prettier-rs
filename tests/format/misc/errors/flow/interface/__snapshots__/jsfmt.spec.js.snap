// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 [babel-flow] format 1`] = `
"\`declare export interface\` is not supported. Use \`export interface\` instead. (1:16)
> 1 | declare export interface A {}
    |                ^"
`;

exports[`snippet: #0 [flow] format 1`] = `
"\`declare export interface\` is not supported. Use \`export interface\` instead. (1:16)
> 1 | declare export interface A {}
    |                ^^^^^^^^^"
`;

exports[`snippet: #1 [babel-flow] format 1`] = `
"Unexpected token, expected "{" (1:13)
> 1 | interface A mixins B {}
    |             ^"
`;

exports[`snippet: #1 [flow] format 1`] = `
"Unexpected identifier, expected the token \`{\` (1:13)
> 1 | interface A mixins B {}
    |             ^^^^^^"
`;

exports[`snippet: #2 [babel-flow] format 1`] = `
"Unexpected token, expected "{" (1:13)
> 1 | interface A implements B {}
    |             ^"
`;

exports[`snippet: #2 [flow] format 1`] = `
"Unexpected identifier, expected the token \`{\` (1:13)
> 1 | interface A implements B {}
    |             ^^^^^^^^^^"
`;

exports[`snippet: #3 [babel-flow] format 1`] = `
"Unexpected token, expected "{" (1:21)
> 1 | declare interface A mixins B {}
    |                     ^"
`;

exports[`snippet: #3 [flow] format 1`] = `
"Unexpected identifier, expected the token \`{\` (1:21)
> 1 | declare interface A mixins B {}
    |                     ^^^^^^"
`;

exports[`snippet: #4 [babel-flow] format 1`] = `
"Unexpected token, expected "{" (1:21)
> 1 | declare interface A implements B {}
    |                     ^"
`;

exports[`snippet: #4 [flow] format 1`] = `
"Unexpected identifier, expected the token \`{\` (1:21)
> 1 | declare interface A implements B {}
    |                     ^^^^^^^^^^"
`;
